crates/ruma-common/src/api/metadata.rs

```diff
@@ -96,6 +101,95 @@ impl Metadata {
 
         VersioningDecision::Unstable
     }
+
+    /// Generate the endpoint URL for this endpoint.
+    pub fn make_endpoint_url(
+        &self,
+        versions: &[MatrixVersion],
+        base_url: &str,
+        path_args: &[&dyn Display],
```

Comment:
I'm not *entirely* comfortable with the fact we're using `dyn Display` like this in a public function... 😬

#1302 is relevant here.